src/include/rados/librados.hpp

```diff
@@ -683,6 +685,9 @@ namespace librados
      * NOTE: this call steals the contents of @param bl.
      */
     int write_full(const std::string& oid, bufferlist& bl);
+    int writesame(const std::string& oid, bufferlist& bl,
+		  size_t data_len, size_t write_len,
```

Comment:
Yes, that would be a valid assumption.
I added the data_len parameter to be consistent with write(const std::string& oid, bufferlist& bl, size_t len, uint64_t off) here. I can remove the parameter if desired.